File: julia/21.jl
function d(n)
	divisorSum = 1
	for i = 2:isqrt(n)
		if n % i === 0
			divisorSum += i
			divisorSum += div(n, i)
		end
	end

	return divisorSum
end

amicableSum = 0
for a = 1:10000
	n1 = d(a)
	n2 = d(n1)

	if n2 === a && n1 !== n2
		global amicableSum += a
	end
end

println(amicableSum)
# 31626

File: solutions/21.c
#include <math.h>
#include <stdio.h>

int d(int n) {
  int divisorSum = 1;
  for (int i = 2; i < sqrt(n); i++) {
    if (n % i == 0) {
      divisorSum += i;
      divisorSum += n / i;
    }
  }

  return divisorSum;
}

int main() {
  int amicableSum = 0;

  for (int a = 1; a <= 10000; a++) {
    int n1 = d(a);
    int n2 = d(n1);

    if (n2 == a && n1 != n2) {
      amicableSum += a;
    }
  }

  printf("%d\n", amicableSum);
  return 0;
}
// 31626
